### ISSUE TITLE
Possible overflow in math operation

### ISSUE THREAD

**Issue by simoatze** [SOLLVE/sollve_vv#380]
Hi,

at this [line ](https://github.com/SOLLVE/sollve_vv/blob/25acf69574d139db161baa45e53b45316d6c871b/tests/4.5/target_teams_distribute/test_target_teams_distribute_reduction_bitand.F90#L61)(and following one) there may be a possible overflow:

The code looks like this:

```
DO y = 1, 32
          result = result + (2**y)
          host_result = host_result + (2**y)
END DO
````

where `2**31` is definitely bigger than `INT_MAX` and both result and host_result are declared as `INTEGER(4)`.
I was wondering if the loop should be limited to `DO y = 1,16` (which is what the [C version](https://github.com/SOLLVE/sollve_vv/blob/master/tests/4.5/target_teams_distribute/test_target_teams_distribute_reduction_bitand.c) of this test does) or those variables should be declared as `INTEGER(8)`.

Thanks!
Simone

**Comment by tmh97** [comment]
Thanks for the help @simoatze ! PR #381 fixes this. I decided to go with limiting `DO y = 1,16` in order to match the C version.

**Comment by simoatze** [comment]
Awesome, thank you!

**Comment by tmh97** [comment]
PR merged #381 , closing issue.
